x-pack/legacy/plugins/task_manager/task.ts

```diff
@@ -216,6 +216,17 @@ export interface TaskInstance {
   ownerId?: string | null;
 }
 
+/**
+ * A task instance that has an id.
+ */
+export interface ExistingTaskInstance extends TaskInstance {
+  /**
+   * The id of the Elastic document that stores this instance's data. This can
+   * be passed by the caller when scheduling the task.
+   */
+  id: string;
```

Comment:
wow, TIL.  You can subclass an interface, changing a super-interface's property from optional to not optional.  Neat-O!

Reply:
That's changed now, as I'm using a new Utility type marking `id` as Required ;)
But the name is now `TaskInstanceWithId`.